### DISCUSSION TITLE
Bootstrapping for CKKS over the conjugate-invariant ring

### DISCUSSION THREAD

**Discussion by caesaretos** [tuneinsight/lattigo#405] [Q&A] [answered]
Hi,

I am interested in bootstrapping for the conjugate-invariant CKKS scheme for a real-number machine learning application. Achieving a factor of 2 in amortized time and plaintext space is appealing for my use case.

It seems that Lattigo does not support bootstrapping for the conjugate-invariant ring CKKS variant. I tried myself a sample code based on the `example/ckks/bootstrapping/main.go` and specified the RingType as `ring.ConjugateInvariant` but I received the following error:
`error message: panic: NewParametersFromLiteral: invalid ring.RingType: must be ring.Standard`. 

Based on this, I assume that it is not supported? If it is not supported, are there any known challenges to implementing this capability?

**Comment by Pro7ech** [comment, marked as answer]
Unfortunately I can confirm that the v4.1 of Lattigo does not support bootstrapping for the Real-CKKS, only for the Complex-CKKS, but we  will keep you posted if we have an update on this front.
